### PR TITLE
Add proper clickable tags to Nhentai

### PR DESCRIPTION
**Pull request by az4521**

### CODE REVIEW

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt [hunk 1]
```diff
@@ -81,6 +82,16 @@ open class NHentai(override val lang: String, private val nhLang: String) : Pars
...
+        val stringBuilder = StringBuilder()
+        val tags = document.select("#tags > div:nth-child(3) > span > a")
+        if (tags.size > 0) {
+            tags.forEach {
+                stringBuilder.append(it.text().replace(Regex("\\(.*\\)"), "").trim())
+                if (it != tags.last())
+                    stringBuilder.append(", ")
+            }
+        }
+        genre = stringBuilder.toString()
```
Comment (arkon) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt]: You can alternatively do something like:

```kotlin
genre = document.select("#tags > .whatever")
        .map { it.text().replace(Regex("\\(.*\\)"), "") }
        .joinToString(", ") { it.trim() }
```

Comment (CarlosEsco) [reply src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt]: yea use the join to String, this is where i would probably remove the duplicates before the join is called.

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt [hunk 2]
```diff
@@ -34,58 +34,26 @@ class NHUtils {
...
+                .joinToString(", ") { it.trim() }            
+        }
+        
         fun getTime(document: Document): Long {
             val timeString = document.toString().substringAfter("datetime=\"").substringBefore("\">").replace("T", " ")
 
             return SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSSZ").parse(timeString).time
         }
-
-        private fun cleanTag(tag: String): String = tag.replace(Regex("\\(.*\\)"), "").trim()
```
Comment (arkon) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt]: Why'd you remove this?

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt [hunk 3]
```diff
@@ -34,58 +34,26 @@ class NHUtils {
...
-            val stringBuilder = StringBuilder()
+        fun getDesc(document: Document): String {
             val parodies = document.select("#tags > div:nth-child(1) > span > a")
+                .map { it.text().replace(Regex("\\(.*\\)"), "") }
+                .joinToString(", ") { it.trim() }
             val characters = document.select("#tags > div:nth-child(2) > span > a")
-            val tags = document.select("#tags > div:nth-child(3) > span > a")
-
-            if (parodies.size > 0) {
-                stringBuilder.append("Parodies: ")
```
Comment (arkon) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt]: What about these headers?

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt [hunk 4]
```diff
@@ -19,6 +19,7 @@ import okhttp3.Request
 import okhttp3.Response
 import org.jsoup.nodes.Document
 import org.jsoup.nodes.Element
+import java.lang.StringBuilder
```
Comment (arkon) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt]: Remove this.

**Comment by CarlosEsco** [comment]
Real comments on commits please

**Comment by az4521** [comment]
I don't even think renaming commits is possible

**Comment by arkon** [comment]
It is, try looking up rebasing, resetting, or squashing.

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt [hunk 5]
```diff
@@ -1,91 +1,45 @@
...
-                stringBuilder.append(cleanTag(it.text()))
-
-                if (it != artists.last())
-                    stringBuilder.append(", ")
-            }
-
-            return stringBuilder.toString()
+            return document.select("#tags > div:nth-child(4) > span > a")
+                .map { cleanTag(it.text()) }
+                .joinToString(", ") { it.trim() }
```
Comment (arkon) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt]: Don't need to trim here if `cleanTag` already handles it.

File: src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt [hunk 6]
```diff
@@ -1,91 +1,45 @@
...
 
-            groups.forEach {
-                stringBuilder.append(cleanTag(it.text()))
-
-                if (it != groups.last())
-                    stringBuilder.append(", ")
-            }
-
-            return if (stringBuilder.toString().isEmpty()) null else stringBuilder.toString()
+            return if (groups.isEmpty()) null else groups
```
Comment (arkon) [inline src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHUtils.kt]: You can use [`ifEmpty`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/if-empty.html).

**Pull request closed without merging**